### ISSUE TITLE
Manufactured solution

### ISSUE THREAD

**Issue by luca-heltai** [ste-dtst/nmpde-project#2]
Compare the solution generated this way with the solution generated in Step 26. Check the accuracy of the method via some manufactured solution.

**Comment by ste-dtst** [comment]
I've done some tests with the functions in `other_files/test_functions.txt` and the approximation looks correct (at least in 1D and 2D), but some cases (e.g. 2.1) show a strange "oscillating" behavior of either the error and the solution at the boundary. Overall, increasing the FE degree (+ tweaking gamma, if needed) improves the $L^2$ error as expected. Tomorrow I will update the readme file with some more details.

**Comment by ste-dtst** [comment]
I have done some more tests and unfortunately I have underestimated a lot the "flickering" effect of the numerical approximation. It seems that the issue does not affect the 1D case, at least not in a relevant way. I have updated the readme file with some thorough analysis.

TL;DR: in 1D, increasing the number of global refinements or the FE degree improves the accuracy; this does not happen in 2D, and I am pretty confident that the same happens in 3D (I've done just one test with dim = 3). Moreover, the numerical solution shows signs of flickering/instability, which can get severe, even when the overall error is $\sim 10^{-2}$ for large parts of the integration interval.

**Comment by ste-dtst** [comment]
A quick (but important) update if you haven't read the readme yet: in the meantime I implemented the assembly with MeshWorker and I am currently testing it. The 1D case works exactly as before, but the 2D case now **_seems_** to run ok and exhibit no flickering. So it may be indeed my fault :( I am even more puzzled, but I will investigate.

**Comment by ste-dtst** [comment]
Investigation completed: as far as I know, I can't reproduce the flickering issue in 2D anymore, even reverting back to the previous code. Also, the bug seemed to affect only the 2D version. It would be very strange, but I can only think I have tested a previous build...

I have updated the readme, hopefully now everything is correct. The 1D and 2D case seem to work fine, the flickering issue is still present in 3D but maybe it's a matter of tweaking the parameters. It's a bit difficult to prove it, though, because the number of DoFs increases pretty quickly.
